src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -302,7 +302,7 @@ private void ReAddReorganisedTransactions(Block? previousBlock)
         private void RemoveProcessedTransactions(Block block)
         {
             Transaction[] blockTransactions = block.Transactions;
-            using ArrayPoolList<Transaction> blobTxsToSave = new(Eip4844Constants.GetMaxBlobsPerBlock());
+            using ArrayPoolList<Transaction> blobTxsToSave = new((int)(block.TargetBlobCount * 2 ?? Eip4844Constants.GetMaxBlobsPerBlock()));
```

Comment:
this throws where EIP7742 is not enabled
and i think to be on the safe side we can just use the block transaction count